using System;

class ValidationException : Exception {}
class SchedulerExcepion : Exception {}

class Post
{
    String content;

    public String Url { get; set; }

    public Post(String content)
    {
        this.content = content;
    }

    public void Validate()
    {
        if (String.IsNullOrEmpty(this.content)
            throw new ValidationException();
    }

    public String GetSlug()
    {
        return this.content.Substring(0, 5);
    }
}

class Scheduler
{
    public void Publish(Post post)
    {
        if (CSharpMagic())
            post.Url = String.Format("https://www.example.com/{0}", post.GetSlug());
    }

    private bool CSharpMagic() {
        return true; // throws SchedulerException
    }
}

class PostHandler
{
    Scheduler scheduler;

    public PostHandler(Scheduler scheduler)
    {
        this.scheduler = scheduler;
    }

    public void Execute(String content)
    {
        var post = new Post(content);
        post.Validate();

        this.scheduler.Publish(post);
        Console.WriteLine("Published: {0}", post.Url);
    }
}

public class Program
{
    public static void Main()
    {
        new PostHandler(new Scheduler()).Execute("content");
    }
}
